### PR TITLE
Add dropdown component prop: pageSize to indicate max display count

### PR DESCRIPTION
**Pull request by gggjtgxing**
#### Pull request checklist

- [ ] Addresses an existing issue: #0000
- [ ] Include a change request file if publishing 
- [ ] New feature, bugfix, or enhancement
  - [ ] Includes tests
- [ ] Documentation update

#### Description of changes

This PR is same with Tim's PR: https://github.com/OfficeDev/office-ui-fabric-react/pull/1550
I have talked with Tim, he is too busy to update this, I'm taking over it. 

#### Focus areas to test

(optional)

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx [hunk 1]
```diff
@@ -272,11 +261,69 @@ export class Dropdown extends BaseComponent<IDropdownInternalProps, IDropdownSta
...
+      }
+    }
+  }
+
   // Render separator
   private _renderSeparator(item: IDropdownOption): JSX.Element {
     let { index, key } = item;
     if (index > 0) {
       return <div
+        ref={ OPTION + item.index }
```
Comment (mdahamiwal) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx]: string refs are legacy and discouraged now https://facebook.github.io/react/docs/refs-and-the-dom.html#legacy-api-string-refs

File: common/changes/master_2017-04-25-02-07.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
+{
+  "changes": [
```
Comment (mdahamiwal) [inline common/changes/master_2017-04-25-02-07.json]: would be great if you could add an example using this prop. It will make it easy to test. Thanks

**Comment by micahgodbolt** [comment]
closing #1550 as this is a replacement of it

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx (hunk 1 above)
Comment (mdahamiwal) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx]: Also do we need refs on seperators if we ignore them during maxheight calculation ? better to use refs for non-separator and calculate height for all of them, this way _calculatemaxheight will get slimmer as well.

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.Props.ts [hunk 3]
```diff
@@ -96,6 +96,11 @@ export interface IDropdownProps extends React.Props<Dropdown> {
   required?: boolean;
 
   /**
+   * Max item per page
+   */
```
Comment (cschleiden) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.Props.ts]: when reading this comment as a consumer, I have no idea what it's supposed to do

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx [hunk 4]
```diff
@@ -272,11 +261,69 @@ export class Dropdown extends BaseComponent<IDropdownInternalProps, IDropdownSta
     }
   }
 
+  private _calculateMaxHeight() {
```
Comment (cschleiden) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx]: this logic looks expensive.. both to maintain and to execute.. finding all the nodes and explicitly measuring them..  again, `List` provides a lot of that (caching, estimation), why duplicate it here?

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx [hunk 5]
```diff
@@ -227,7 +216,7 @@ export class Dropdown extends BaseComponent<IDropdownInternalProps, IDropdownSta
           onDismiss={ this._onDismiss }
           onPositioned={ this._onPositioned }
         >
-          <div style={ { width: this._dropDown.clientWidth - 2 } }>
+          <div style={ { width: this._dropDown.clientWidth - 2, maxHeight: calculatedHeight === 0 ? undefined : calculatedHeight } }>
```
Comment (cschleiden) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx]: If this is the main point of the PR, why not just have users pass in the `maxHeight`?

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx (hunk 4 above)
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx]: I totally agree. We should not do this.

**Comment by dzearing** [comment]
I'm closing this; there are too many bad things here. 

1. The prop is very unclear; it actually is measuring X items and not limiting the options in the dropdown, so it's misleading.
2. Measuring is basically a bad idea here, and the way its done is going to introduce perf issues.
3. Using string refs is not ideal; we should avoid this.

It may be more appropriate to allow css classnames to be injected on the options container, that might allow the caller to own this logic.

**Pull request closed without merging**